src/transformStateWithClones.js

```diff
@@ -7,7 +7,60 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateArr = [];
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+    const newState = stateArr.length
+      ? { ...stateArr[stateArr.length - 1] }
+      : { ...state };
+
+    switch (type) {
+      case 'addProperties':
+        stateArr.push(Object.assign(newState, extraData));
```

Comment:
You're pushing the `newState` into `stateArr` inside the `switch` block. It's better to do this once after the `switch` block to avoid repetition and to ensure that the same action isn't performed multiple times. [DONT REPEAT YOURSELF]